### DISCUSSION TITLE
Best approach to support both APNs and FCM

### DISCUSSION THREAD

**Discussion by jimmymjing** [airsdk/apm#96] [Q&A] [answered]
Hi, is it possible to just maintain a single APM repo to support both APNs for iOS and FCM for Android? 

I listed the push notification related ANEs but not found that `allservices` ANE, and I can't have both `pushnotifications` and `fcm` in the ane folder (the APM popup an alert saying pushnotifications already installed. I assume they have the same signature and may have conflicts) 

Do I have to have 2 separated APM repos? Thanks!

**Comment by marchbold** [comment, marked as answer]
The base push notifications extension should support FCM on Android and APNS on iOS. Just use the default extension and not one of the variants.

**Reply by marchbold** [reply]
Must say this is one area that we are interested in feedback on, regarding the variants and how to best document them. So please let me know if you have any feedback around this.

**Reply by httpwebmedia** [reply]
It contradicts what you wrote in the documentation here : https://docs.airnativeextensions.com/docs/pushnotifications/firebase/firebase-cloud-messaging

I've tried to use APM (which I think is a very good idea I must say this is a tool I dreamed about) but I receive an error : 
CONFLICT :: fatal error : found [4] conflicting packages
CONFLICT :: com.distriqt.playservices.Base
CONFLICT :: ├── 18.0.1 required by: com.distriqt.Adverts@13.5.0
CONFLICT :: └── 17.5.1 required by: com.distriqt.PushNotifications@10.4.5
CONFLICT :: com.distriqt.playservices.CloudMessaging
CONFLICT :: ├── 17.0.2 required by: com.google.firebase.core@29.0.4
CONFLICT :: └── 16.0.0 required by: com.distriqt.PushNotifications@10.4.5
CONFLICT :: androidx.constraintlayout
CONFLICT :: ├── 2.1.3 required by: com.distriqt.Adverts@13.5.0
CONFLICT :: └── 1.1.3 required by: com.distriqt.PushNotifications@10.4.5
CONFLICT :: com.google.firebase.core
E::ProcessQueue::Error #1009
install :: Error #1009

Also I think it might be related to the recent issue I got, check this out  : 
https://github.com/distriqt/ANE-AndroidSupport/issues/18

**Reply by marchbold** [reply]
We are currently in midst of a major update of all our extensions so the latest versions of all of them are not currently compatible as apm is correctly reporting here. Your options are to wait until the update is complete (end of next week) or to install the previous version of the extensions.
